lib/semantics/resolve-names.cc

```diff
@@ -3141,12 +3211,8 @@ bool ResolveNamesVisitor::Pre(const parser::MainProgram &x) {
   } else {
     PushScope(Scope::Kind::MainProgram, nullptr);
   }
-  if (auto &subpPart{
-          std::get<std::optional<parser::InternalSubprogramPart>>(x.t)}) {
-    subpNamesOnly_ = SubprogramKind::Internal;
-    Walk(*subpPart);
-    subpNamesOnly_ = std::nullopt;
-  }
+  auto &subpPart{std::get<std::optional<parser::InternalSubprogramPart>>(x.t)};
```

Comment:
should subp be spelled out as subprogram?

Reply:
I think that is often too verbose for something that is not part of an API. But if the consensus is we should always spell it out I'll change it.